### DISCUSSION TITLE
Z-Wave Controller migration flow to add and migrate to a secondary Z-Wave Controller (add and promote secondary controller to primary controller)?

### DISCUSSION THREAD

**Discussion by Hedda** [zwave-js/zwave-js#7268] [Feature Requests]
### Description

Please consider implementing all dependecis and prerequicites to then add simply-to-use migration flow that will allow users to both add secondary Z-Wave Controller (Z-Wave 800 Series)and migrate to that secondary Z-Wave Controller by promoting that secondary controller into primary controller and assiging the new controller node ID number one (and either demoting or or removing your old controller).

The point is that your new controller will totally take over your existing Z-Wave network and you do not need to re-pair any devices, with the end result being that your old controller is no longer primary controller and Node 1 but instead your new controller is.

Would be great if an automated migration flow could be added that backup the old controller then automatically takes you through the steps to add your new Z-Wave Controller as a secondary controller before promoting it and the re-assign node ID.

Users can already achieve that by using Silabs Z-Wave PC Controller toolkit from Silicon Labs Simplicity Studio development software. See that secondary controller method and promotion process demonstrated in this video here:

* https://www.youtube.com/watch?v=lktIFOcg7GI

Note that the process might differ slightly if user have an old controller with older firmware on the old primary controller, see:

* https://github.com/zwave-js/zwave-js-ui/issues/3828

Back-story; historically if a user wants to move/migrate to a different Z-Wave Controller then the easiest approach/method is to do an NVM backup to file from the old/original controller and then restore that backup to a new Z-Wave Controller, but that is not supported or at least not recommended when you want to move/migrate from a Z-Wave 500 series controller or Z-Wave 700 series controller to a Z-Wave 800 series controller. An alternative approach/method is to instead add a secondary Z-Wave Controller to your existing Z-Wave network, (making your network contain both a primary controller and a secondary controller at the same time), and then you promote your secondary controller into a primary controller (and demote or remove your old controller).

**Comment by AlCalzone** [comment]
The downside of that approach (although it's the "intended" one) is that
a) the other controller software needs to support promoting another controller to primary
b) the node ID of the new primary won't be 1, unless you jump trough hoops with virtual nodes.

I have something else planned already: https://github.com/zwave-js/node-zwave-js/issues/7096
